Ginger/GingerCoreNET/Run/Context.cs

```diff
@@ -112,6 +131,144 @@ public TargetBase Target
             }
         }
 
+        private string mTargetApplication;
+        public string TargetApplication
+        {
+            get
+            {
+                return mTargetApplication;
+            }
+            set
+            {
+                if (mTargetApplication != value)
+                {
+                    mTargetApplication = value;
+                    OnPropertyChanged(nameof(TargetApplication));
+                }
+            }
+        }
+
+        private ApplicationAgent mAppAgent;
+        public ApplicationAgent AppAgent
+        {
+            get
+            {
+                return mAppAgent;
+            }
+            set
+            {
+                if (mAppAgent != value)
+                {
+                    mAppAgent = value;
+                    mAppAgent.PropertyChanged += AppAgent_PropertyChanged;
+                    Agent = mAppAgent.Agent;
+                    OnPropertyChanged(nameof(AppAgent));
+                }
+            }
+        }
+        
+        private Agent mAgent;
+        public Agent Agent
+        {
+            get
+            {
+                return mAgent;
+            }
+            set
+            {
+                if (mAgent != value)
+                {
+                    mAgent = value;
+                    mAgent.PropertyChanged += Agent_PropertyChanged;
+                    OnPropertyChanged(nameof(Agent));
+                }
+            }
+        }
+        
+        private string mAgentStatus;
+        public string AgentStatus
+        {
+            get
+            {
```

Comment:
no need AgentStatus we can pull it from Agent

Reply:
It is required, since to notify to child page, it is just used to get the notification of the status changed